lib/longtail_platform.c

```diff
@@ -1091,13 +1102,22 @@ int Longtail_Write(HLongtail_OpenFile handle, uint64_t offset, uint64_t length,
     OVERLAPPED WriteOp;
     memset(&WriteOp, 0, sizeof(WriteOp));
 
-    WriteOp.Offset  = (DWORD)(offset & 0xffffffff);
-    WriteOp.OffsetHigh = (DWORD)(offset >> 32);
-
-    if (FALSE == WriteFile(h, input, (DWORD)length, 0, &WriteOp))
+    const char* cur = (const char*)input;
+    const char* end = cur + length;
+    while (cur < end)
     {
-        return Win32ErrorToErrno(GetLastError());
+        WriteOp.Offset  = (DWORD)(offset & 0xffffffff);
+        WriteOp.OffsetHigh = (DWORD)(offset >> 32);
+
+        const DWORD part_size = LONGTAIL_WIN32_PART_SIZE(cur, end);
+        if (FALSE == WriteFile(h, cur, part_size, 0, &WriteOp))
```

Comment:
We should make use of the fourth parameter - `lpNumberOfBytesWritten` to determine how many bytes we actually read.